### PR TITLE
LIVE-886 (Market): add sparkline variation + fix issue in countervalue initial value

### PR DESCRIPTION
**Pull request by LFBarreto**
- (Market): add svg sparkline variation + fix issue in countervalue initial setting

## Context (issues, jira)

[LIVE-886]

## Expectations

- [ ] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [ ] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

[LIVE-886]: https://ledgerhq.atlassian.net/browse/LIVE-886?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/market/api/api.ts [hunk 1]
```diff
@@ -83,6 +83,7 @@ function sparklineAsSvgData(points: number[]): SparklineSvgData {
       })
       .join(" "),
     viewBox: `0 0 ${totalXSteps} ${sparklineYHeight + 3}`,
+    isPositive: points[0] <= points[points.length - 1],
```
Comment (haammar-ledger) [inline src/market/api/api.ts:86]: Can `points` be empty here? (noob question I have 0 context on market feature)

Comment (LFBarreto) [reply src/market/api/api.ts:88]: good point should be handled now :+1:

File: src/market/api/api.ts [hunk 2]
```diff
@@ -52,6 +52,8 @@ const matchSearch =
   };
 
 function distributedCopy(items: number[], n: number): number[] {
+  if (!items) return [];
+  if (items.length <= n) return items;
```
Comment (haammar-ledger) [inline src/market/api/api.ts:56]: This safety looks good for the `distributedCopy` function :+1: but does it help for the `sparklineAsSvgData` function below?

Comment (LFBarreto) [reply src/market/api/api.ts:56]: its always called with data provided by this function so in this case yes :sweat_smile:

Comment (LFBarreto) [reply src/market/api/api.ts:56]: line 185
```js

 sparklineIn7d: currency?.sparkline_in_7d?.price
        ? sparklineAsSvgData(
            distributedCopy(currency.sparkline_in_7d.price, 6 * 7)
          ) // keep 6 points per day
        : null,
```

**Review by ghost — APPROVED** [review]

**Pull request merged**